### PR TITLE
smaller journal fixes

### PR DESCRIPTION
**Pull request by poettering**
A couple of fixes that allow us to deal nicer with partly corrupted journal files. In particular this makes us handle corrupted files nicer both when we linearly iterate through them and when we seek through them.

### CODE REVIEW

File: src/journal/journal-file.c [hunk 1]
```diff
@@ -2062,9 +2067,14 @@ static int generic_array_bisect(
...
-
-                                r = test_object(f, p, needle);
+                                        r = -EBADMSG;
+                                else
+                                        r = test_object(f, p, needle);
+                                if (r == -EBADMSG) {
+                                        log_debug_errno(r, "Encountered invalid entry while bisecting, cutting algorithm short. (2)");
+                                        right = n = i;
+                                        continue;
+                                }
```
Comment (zonque) [inline src/journal/journal-file.c:2077]: Hmm, this is hard to read, but maybe Im confused. Why not move this `if (r == -EBADMSG)` block into the `if (p <= 0)` condition above?

File: man/systemd.exec.xml [hunk 2]
```diff
@@ -470,6 +469,10 @@
         similar to the same option of
         <varname>StandardInput=</varname>.</para>
 
+        <para>If the standard output (or error output, see below) of a unit is connected with the journal, syslog or
+        the kernel log buffer the unit will implicitly gain a dependency of type <varname>After=</varname> on
```
Comment (zonque) [inline man/systemd.exec.xml:473]: A comma after ~~"unit"~~ "buffer" would give the text a better flow.

**Comment by zonque** [comment]
I don't know much about the internals of journald, but this looks all sane to me except for the two minor nits I pointed out.

File: src/journal/journal-file.c (hunk 1 above)
Comment (poettering) [inline src/journal/journal-file.c:2077]: test_object() might return -EBADMSG too. Or in other words: p <= tells us the offset is bogus, and test_object() tells us whether the object behind the offset is bogus. For us the distinction doesn't matter much, in either case, we want to cut the bisection short.

Comment (zonque) [inline src/journal/journal-file.c:2077]: Make sense, thanks!

File: man/systemd.exec.xml (hunk 2 above)
Comment (poettering) [inline man/systemd.exec.xml:473]: will fix in a later PR. thanks!

**Comment by vcaputo** [comment]
lfad

**Comment by poettering** [comment]
@vcaputo hmm, "lfad", what does that mean?

**Comment by vcaputo** [comment]
looks fine and dandy

**Comment by poettering** [comment]
ahhh, we used "lgtm" so far for that, but "lfad" lfad to me, too ;-)

**Pull request merged**